### ISSUE TITLE
nativeEnqueuer.processNativeClasses slows down --analyze-only

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/sdk#19954]
I'm currently working on extracting type information from dart2js as fast as possible (see https://codereview.chromium.org/385573002/).

When running poi.dart on hello_html.dart, the compiler reports:

Info: Total compile-time 2094msec; unaccounted 113msec

If I apply this diff:

--- a/dart/sdk/lib/_internal/compiler/implementation/compiler.dart
+++ b/dart/sdk/lib/_internal/compiler/implementation/compiler.dart
@@ -1452,7 +1452,6 @­@ abstract class Compiler implements DiagnosticListener {
&nbsp;&nbsp;&nbsp;}
&nbsp;
&nbsp;&nbsp;&nbsp;void processQueue(Enqueuer world, Element main) {
-    world.nativeEnqueuer.processNativeClasses(libraryLoader.libraries);
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if (main != null) {
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FunctionElement mainMethod = main;
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if (mainMethod.computeSignature(this).parameterCount != 0) {

The compiler reports:

Info: Total compile-time 888msec; unaccounted 90msec

In other words, for --analyze-all and fast type checking in Try Dart, it might be worth investigating if nativeEnqueuer.processNativeClasses can be made more on-demand.

I've run poi.dart with this command:

./sdk/bin/dart -p$BUILD_DIRECTORY/ReleaseIA32/packages/ site/try/poi/poi.dart hello_html.dart 129

hello_html.dart is:

// Go ahead and modify this example.

import &quot;dart:html&quot;;

var greeting = &quot;Hello, World!&quot;;

// Displays a greeting.
void main() {
&nbsp;&nbsp;// This example uses HTML to display the greeting and it will appear
&nbsp;&nbsp;// in a nested HTML frame (an iframe).
&nbsp;&nbsp;document.body.append(new HeadingElement.h1()..appendText(greeting));
}

**Comment by kevmoo** [comment]
We're going to deprecate registerElement
